### PR TITLE
PR: FE/IN Protected Route

### PR DESCRIPTION
**Pull request by ababber**
### What this PR does (required):
- added protected route hoc for dashboard, edit-profile, or any page that requires the user to login

### Screenshots / Videos (front-end only):

https://user-images.githubusercontent.com/18492009/142589559-947d3fde-7d8f-489d-9e69-3ecf37c6c7a4.mov

### Any information needed to test this feature (required):
- launch app, attempt to manually access routes that require the user to be logged in

### CODE REVIEW

File: client/src/components/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Component: React.ComponentType;
+}
+
+const ProtectedRoute = ({ Component, ...rest }: IProtectedRoute): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  return (
+    <Route
+      render={(props: RouteComponentProps) => {
+        //for demonstrating performance issue
+        console.log('rendering...');
```
Comment (ezeilo-su) [inline client/src/components/ProtectedRoute.tsx:15]: Please get rid of these two lines :smiley:

**Review by ezeilo-su — COMMENTED** [review]
Looks good! Just a minor comment. Please note that it is required that for every PR you raise link the issue being addressed. You can check this [doc](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for a guide :+1: 
Happy coding!

File: client/src/components/ProtectedRoute.tsx (hunk 1 above)
Comment (ababber) [reply client/src/components/ProtectedRoute.tsx:15]: I left this there as part of the initial PR comment about the performance issue. In the second video that demonstrates the performance issue, all protected routes continuously render. I will remove it once the issue has been resolved.

File: client/src/components/ProtectedRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+interface IProtectedRoute extends RouteProps {
+  Component: React.ComponentType;
+}
+
+const ProtectedRoute = ({ Component, ...rest }: IProtectedRoute): JSX.Element => {
+  const { loggedInUser } = useAuth();
+  return (
+    <Route
+      render={(props: RouteComponentProps) => {
+        if (loggedInUser !== null) {
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute.tsx:14]:
```suggestion
        if (!loggedInUser) {
```

File: client/src/components/ProtectedRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Route, RouteComponentProps, RouteProps, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+interface IProtectedRoute extends RouteProps {
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute.tsx:5]:
```suggestion
interface ProtectedRoute extends RouteProps {
```
It is recommended not to prefix interfaces with 'I' :)

File: client/src/pages/Unauthorized/useStyles.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  root: {
+    minHeight: '100vh',
+    '& .MuiInput-underline:before': {
+      borderBottom: '1.2px solid rgba(0, 0, 0, 0.2)',
+    },
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  boxWrapper: {
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/pages/Unauthorized/useStyles.ts:13]: Most of these styles can be applied to the Box component directly

Eg: `<Box display="flex" alignItems="center"...>`

**Review by rajivtitus — APPROVED** [review]
Great work adding in an unauthorized page :)

Just a couple of minor comments to address before merging, thanks

**Pull request merged**